cmd/frontend/internal/auth/accessrequest/handlers.go

```diff
@@ -0,0 +1,121 @@
+	"github.com/sourcegraph/sourcegraph/internal/usagestats"
+)
+
+// HandleRequestAccess handles submission of the request access form.
+func HandleRequestAccess(logger log.Logger, db database.DB) http.HandlerFunc {
+	logger = logger.Scoped("HandleRequestAccess", "request access request handler")
+	return func(w http.ResponseWriter, r *http.Request) {
+		if handleEnabledCheck(logger, w) {
+			return
+		}
+		// Check whether builtin signup is enabled.
+		builtInAuthProvider, _ := userpasswd.GetProviderConfig()
+		if builtInAuthProvider != nil && builtInAuthProvider.AllowSignup {
+			http.Error(w, "Use sign up instead.", http.StatusConflict)
+			return
+		}
+		handleRequestAccess(logger, db, w, r)
+	}
+}
+
+// HandleRequestAccessEnabledCheck checks whether request access experimental feature is explicitly disabled
+func handleEnabledCheck(logger log.Logger, w http.ResponseWriter) (handled bool) {
+	experimentalFeatures := conf.Get().ExperimentalFeatures
+	if experimentalFeatures != nil && experimentalFeatures.AccessRequests != nil && !experimentalFeatures.AccessRequests.Enabled {
+		logger.Error("Request access is disabled.")
+		http.Error(w, "Request access is disabled.", http.StatusConflict)
+		return true
+	}
+
+	// test whether
+	return false
+}
+
+type requestAccessData struct {
+	Name           string `json:"name"`
+	Email          string `json:"email"`
+	AdditionalInfo string `json:"additionalInfo"`
+}
+
+// handleRequestAccess handles submission of the request access form.
+func handleRequestAccess(logger log.Logger, db database.DB, w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		http.Error(w, fmt.Sprintf("unsupported method %s", r.Method), http.StatusBadRequest)
+		return
+	}
+	var data requestAccessData
+	if err := json.NewDecoder(r.Body).Decode(&data); err != nil {
+		http.Error(w, "could not decode request body", http.StatusBadRequest)
+		return
+	}
+
+	const defaultErrorMessage = "Request access failed unexpectedly."
+
+	if err := userpasswd.CheckEmailFormat(data.Email); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Create the access_request.
+	_, err := db.AccessRequests().Create(r.Context(), &types.AccessRequest{
+		Name:           data.Name,
+		Email:          data.Email,
+		AdditionalInfo: data.AdditionalInfo,
+	})
+	if err != nil {
+		var (
+			message    string
+			statusCode int
+		)
+		switch {
+		case database.IsAccessRequestUserWithEmailExists(err):
+			// TODO: clarify how to handle this error, since this will be shown to unauthenticated user
+			message = "A user with this email already exists."
+			statusCode = http.StatusConflict
+		case database.IsAccessRequestWithEmailExists(err):
+			// TODO: clarify how to handle this error, since this will be shown to unauthenticated user
+			message = "A access request was already created previously."
+			statusCode = http.StatusConflict
+		case errcode.PresentationMessage(err) != "":
```

Comment:
```suggestion
		case message := errcode.PresentationMessage(err); message != "":
```

Hmm, not sure if it works, but maybe it does?

Reply:
Doesn't seem to work :(, does not understand syntax.